File: frontend/ReadBuddy/ReadBuddy/Models/Quiz/Question.cs
// ReadBuddy\Models\Quiz\Question.cs
using ReadBuddy.Models.Extraction;
using ReadBuddy.Models.TTS;
using ReadBuddy.Services;
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace ReadBuddy.Models.Quiz
{
    public class Question
    {
        [JsonPropertyName("id")]
        public string Id { get; private set; }

        [JsonPropertyName("user_id")]
        public string UserId { get; private set; }

        [JsonPropertyName("content")]
        public string Content { get; private set; }

        [JsonPropertyName("answers")]
        public List<Answer> Answers { get; private set; }

        [JsonPropertyName("correct_answer_id")]
        public string CorrectAnswerId { get; private set; }

        [JsonPropertyName("audio_zip_urls")]
        public List<string> AudioZipUrls { get; private set; }

        public TTSResult? ttsResult { get; set; } = null;

        public bool withAudio => this.ttsResult != null && this.ttsResult.AudioZipUrls.Count > 0;

        public Question(
            string userId,
            string content,
            List<Answer> answers,
            string correctAnswerId,
            List<string> audioZipUrls,
            string id)
        {
            UserId = userId;
            Id = id;
            Content = content;
            Answers = answers;
            CorrectAnswerId = correctAnswerId;
            AudioZipUrls = audioZipUrls;
        }
    }
}

File: backend/app/models/question.py
# app/models/question.py

from typing import List
from app.models.answer import Answer
import uuid

class Question:
    """
    Represents a question in a quiz, containing multiple answers and the correct answer.
    """
    def __init__(
        self,
        user_id: str,
        content: str,
        answers: List[Answer],
        correct_answer_id: str,
        audio_zip_urls: List[str] = None,
        id: str = None
    ):
        self.user_id = user_id
        self.id = f"q-{uuid.uuid4().hex[:12]}" if not id else id
        self.content = content
        self.answers = answers
        self.correct_answer_id = correct_answer_id
        self.audio_zip_urls = audio_zip_urls if audio_zip_urls else []
        
    @staticmethod
    def from_dict(user_id: str, data: dict) -> "Question":
        """
        Static method to create a Question object from a dictionary.
        """
        options_dict = data["options"]
        correct_option_letter = data["correct_answer"]
        answers = []
        correct_answer_id = None

        for letter, content in options_dict.items():
            answer = Answer(user_id=user_id, content=content)
            if letter == correct_option_letter:
                correct_answer_id = answer.id
            answers.append(answer)

        return Question(
            user_id=user_id,
            content=data["question"],
            answers=answers,
            correct_answer_id=correct_answer_id
        )

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "user_id": self.user_id,
            "content": self.content,
            "answers": [answer.to_dict() for answer in self.answers],
            "correct_answer_id": self.correct_answer_id,
            "audio_zip_urls": self.audio_zip_urls
        }